src/app/core/messages/messages.service.js

```diff
@@ -40,6 +40,10 @@ module.exports = function() {
 
 	function dismissMessage(messageIds, user, headers) {
 		const dismissedMessagePromises = [];
+		const saveDismissedMessagePromise = (dismissedMessage) => {
+			const dismissedMessagePromise = dismissedMessage.save();
+			dismissedMessagePromises.push(dismissedMessagePromise);
+		};
 
```

Comment:
SonarQube doesn't like it when we define functions inside loops, so I lifted it out.